steelscript/appresponse/core/reports.py

```diff
@@ -379,22 +397,10 @@ def _cast_number(self, result, source_name):
 
         logger.debug("Converting string in records into integer/float")
 
-        columns = self.appresponse.reports.sources[source_name]['columns']
-        functions = [lambda x: x] * len(result['columns'])
-
-        for i, col in enumerate(result['columns']):
-            if columns[col]['type'] in ['integer', 'timestamp']:
-                functions[i] = (lambda x: None if x == 'NULL'
-                                else int(x) if x.isdigit() else x)
-            elif columns[col]['type'] in ('number', 'duration'):
-                functions[i] = (lambda x: None if x == 'NULL'
-                                else float(x) if
-                                x.replace('.', '', 1).isdigit() else x)
-
         # operate on each column, then zip back into list of tuples
         datacols = []
         for i, c in enumerate(zip(*result['data'])):
-            datacols.append(map(functions[i], c))
+            datacols.append(map(None, c))
```

Comment:
I can't replicate the exceptions using your example command.  Go ahead and revert the change for these value conversions and let's explore in a different issue.

Reply:
I'll revert and open an issue.

You should be able to see it if you issue this command:

(venv) C02VK0R2G8WL-MAC:examples spendleton$ python /Users/spendleton/git/steelscript-appresponse/examples/general_report.py [yourappliance] -u [user] -p [pass] --sourcename=wtapages --valuecolumns=web.url,web.time,web.start_time --topbycolumns=web.start_time --timerange='last 60 sec' --limit=10
/Users/spendleton/steelscript/venv/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:730: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html (This warning will only appear once by default.)
  InsecureRequestWarning)
Traceback (most recent call last):
  File "/Users/spendleton/git/steelscript-appresponse/examples/general_report.py", line 148, in <module>
    app.run()
  File "/Users/spendleton/steelscript/venv/lib/python2.7/site-packages/steelscript/common/app.py", line 35, in run
    self.parse(sys.argv[1:])
  File "/Users/spendleton/steelscript/venv/lib/python2.7/site-packages/steelscript/commands/steel.py", line 306, in parse
    self.main()
  File "/Users/spendleton/git/steelscript-appresponse/examples/general_report.py", line 132, in main
    report.run()
  File "/Users/spendleton/steelscript/venv/lib/python2.7/site-packages/steelscript/appresponse/core/reports.py", line 435, in run
    source_name)
  File "/Users/spendleton/steelscript/venv/lib/python2.7/site-packages/steelscript/appresponse/core/reports.py", line 399, in _cast_number
    columns = self.appresponse.reports.sources[source_name]['columns']
KeyError: 'wtapages'

Reply:
I believe the KeyError is caused by the lack of --keycolumns on the command. Not all queries require a --keycolumns option. But I am not sure what _cast_number does.